alf/networks/encoding_networks.py

```diff
@@ -337,6 +330,126 @@ def forward(self, inputs, state=()):
+            activation (nn.functional): activation for hidden layers
+            kernel_initializer (Callable): initializer for all the layers.
+            output_activation (nn.functional): activation for the output layer.
+                Usually our image inputs are normalized to [0, 1] or [-1, 1],
+                so this function should be ``torch.sigmoid`` or
+                ``torch.tanh``.
+            name (str):
+        """
+        super().__init__(
+            input_tensor_spec=TensorSpec((input_size, )), name=name)
+
+        assert isinstance(transconv_layer_params, tuple)
+        assert len(transconv_layer_params) > 0
+
+        self._preprocess_fc_layers = nn.ModuleList()
+        if preprocess_fc_layer_params is not None:
+            for size in preprocess_fc_layer_params:
+                self._preprocess_fc_layers.append(
+                    layers.ParallelFC(
+                        input_size,
+                        size,
+                        n,
+                        activation=activation,
+                        kernel_initializer=kernel_initializer))
+                input_size = size
+
+        start_decoding_size = common.tuplify2d(start_decoding_size)
+        # pytorch assumes "channels_first" !
+        self._start_decoding_shape = [
+            start_decoding_channels, start_decoding_size[0],
+            start_decoding_size[1]
+        ]
+        self._preprocess_fc_layers.append(
+            layers.ParallelFC(
+                input_size,
+                np.prod(self._start_decoding_shape),
+                n,
+                activation=activation,
+                kernel_initializer=kernel_initializer))
+
+        self._transconv_layer_params = transconv_layer_params
+        self._transconv_layers = nn.ModuleList()
+        in_channels = start_decoding_channels
+        for i, paras in enumerate(transconv_layer_params):
+            filters, kernel_size, strides = paras[:3]
+            padding = paras[3] if len(paras) > 3 else 0
+            if same_padding:  # overwrite paddings
+                kernel_size = common.tuplify2d(kernel_size)
+                padding = ((kernel_size[0] - 1) // 2,
+                           (kernel_size[1] - 1) // 2)
+            act = activation
+            if i == len(transconv_layer_params) - 1:
+                act = output_activation
+            self._transconv_layers.append(
+                layers.ParallelConvTranspose2D(
+                    in_channels,
+                    filters,
+                    kernel_size,
+                    n,
+                    activation=act,
+                    kernel_initializer=kernel_initializer,
+                    strides=strides,
+                    padding=padding))
+            in_channels = filters
+        self._n = n
+
+    def forward(self, inputs, state=()):
+        """Returns an image of shape ``(B,C,H,W)``. The empty state just keeps the
```

Comment:
shape comment is incorrect

Reply:
Corrected

Reply:
Inputs can be either one or multiple replicas here?

Reply:
That's correct. Added more comments and tests on this.